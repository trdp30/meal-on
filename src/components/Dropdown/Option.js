import React, { memo } from 'react';
import { components } from 'react-select';

const Option = props => {
  const {
    selectProps: { OptionComponent, primaryKey, labelKey },
  } = props;
  return (
    <components.Option {...props}>
      {OptionComponent ? (
        React.cloneElement(OptionComponent, {
          key: props.data[String(primaryKey)],
          data: props.data,
          primaryKey,
          labelKey,
          ...props,
        })
      ) : (
        <>{props.children}</>
      )}
    </components.Option>
  );
};

export default memo(Option);
