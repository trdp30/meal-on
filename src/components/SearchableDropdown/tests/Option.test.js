import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import Option from '../Option';

describe('Option', () => {
  const props = {
    selectProps: {
      OptionComponent: null,
      primaryKey: 'id',
      labelKey: 'name',
    },
    getStyles: jest.fn(),
    getClassNames: jest.fn(),
    cx: jest.fn(),
    data: {
      id: 1,
      name: 'Option 1',
      email: 'option1@example.com',
    },
  };

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<Option {...props}>Child Content</Option>);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const { container } = render(<Option {...props}>Child Content</Option>);
    expect(container).toMatchSnapshot();
  });
  it('renders correctly', () => {
    const { getByText } = render(<Option {...props}>Child Content</Option>);

    // Assert that the default content is rendered
    expect(getByText('Child Content')).toBeInTheDocument();
  });
});
