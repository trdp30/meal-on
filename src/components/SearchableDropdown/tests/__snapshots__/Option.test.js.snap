// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Option Should render and match the snapshot 1`] = `
<div>
  <div
    class="css-0"
  >
    Child Content
  </div>
</div>
`;
